extension/agenthealth/extension.go

```diff
@@ -24,11 +25,31 @@ var _ awsmiddleware.Extension = (*agentHealth)(nil)
 func (ah *agentHealth) Handlers() ([]awsmiddleware.RequestHandler, []awsmiddleware.ResponseHandler) {
 	var responseHandlers []awsmiddleware.ResponseHandler
 	requestHandlers := []awsmiddleware.RequestHandler{useragent.NewHandler(ah.cfg.IsUsageDataEnabled)}
-	if ah.cfg.IsUsageDataEnabled {
-		req, res := stats.NewHandlers(ah.logger, ah.cfg.Stats)
-		requestHandlers = append(requestHandlers, req...)
-		responseHandlers = append(responseHandlers, res...)
+
+	if !ah.cfg.IsUsageDataEnabled {
+		ah.logger.Debug("Usage data is disabled, skipping stats handlers")
+		return requestHandlers, responseHandlers
+	}
+
+	statusCodeEnabled := ah.cfg.IsStatusCodeEnabled
+
+	var statsResponseHandlers []awsmiddleware.ResponseHandler
+	var statsRequestHandlers []awsmiddleware.RequestHandler
+	var statsConfig agent.StatsConfig
+	var agentStatsEnabled bool
+
+	if ah.cfg.Stats != nil {
+		statsConfig = *ah.cfg.Stats
+		agentStatsEnabled = true
+	} else {
+		agentStatsEnabled = false
 	}
+
+	statsRequestHandlers, statsResponseHandlers = stats.NewHandlers(ah.logger, statsConfig, statusCodeEnabled, agentStatsEnabled)
```

Comment:
nit: don't need to define variables
```suggestion
	var statsConfig agent.StatsConfig
	var agentStatsEnabled bool

	if ah.cfg.Stats != nil {
		statsConfig = *ah.cfg.Stats
		agentStatsEnabled = true
	} else {
		agentStatsEnabled = false
	}

	statsRequestHandlers, statsResponseHandlers := stats.NewHandlers(ah.logger, statsConfig, statusCodeEnabled, agentStatsEnabled)
```
(then you can remove the conditionals from @jefchien's changes)